src/core_plugins/kibana/public/dashboard/reducers/embeddables.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import _ from 'lodash';
+import { Reducer } from 'redux';
+
+import {
+  EmbeddableActionTypeKeys,
+  EmbeddableErrorActionPayload,
+  EmbeddableIsInitializedActionPayload,
+  EmbeddableIsInitializingActionPayload,
+  PanelActionTypeKeys,
+  SetStagedFilterActionPayload,
+} from '../actions';
+import {
+  EmbeddableReduxState,
+  EmbeddablesMap,
+  PanelId,
+} from '../selectors/types';
+
+const embeddableIsInitializing = (
+  embeddables: EmbeddablesMap,
+  payload: EmbeddableIsInitializingActionPayload
+): EmbeddablesMap => ({
+  ...embeddables,
+  [payload]: {
+    error: undefined,
```

Comment:
__nit:__ I believe if you define `error` and `stagedFilter` as optional properties you won't have to define them here at all:
```ts
error?: string | object;
stagedFilter?: object;
```